skyportal/models.py

```diff
@@ -547,13 +547,30 @@ def token_groups(self):
 Token.groups = token_groups
 
 
-class Obj(Base, ha.Point):
+class CommentableBase(Base):
+    """
+    A base class for all other classes that can be assigned comments and annotations.
+    E.g., Obj, Spectrum, Photometry, Thumbnails
+    """
+
+    # id should be defined by the baselayer BaseMixin
+    commentable_type = sa.Column(sa.String)
+
+    __mapper_args__ = {
+        'polymorphic_identity': 'commentable_base',
+        'polymorphic_on': 'commentable_type',
+    }
+
+
+class Obj(CommentableBase, ha.Point):
     """A record of an astronomical Object and its metadata, such as position,
     positional uncertainties, name, and redshift."""
 
     update = public
 
-    id = sa.Column(sa.String, primary_key=True, doc="Name of the object.")
+    # id = sa.Column(sa.String, primary_key=True, doc="Name of the object.")
+    name = sa.Column(sa.String, primary_key=True, doc="Name of the object.")
+    id = sa.Column(sa.Integer, sa.ForeignKey('commentablebases.id'), primary_key=True)
```

Comment:
This will just be inherited from `Base`, like all other `Base`-based models, so no need for this

Reply:
And I don't think we should be changing `Obj.id` to an int and moving the current `Obj.id` to `Obj.name`